### ISSUE TITLE
`NormLayer` should ignore padded frames for mean/variance calculation

### ISSUE THREAD

**Issue by albertz** [rwth-i6/returnn#575]
When `axes` includes some dynamic axis with sequence length.

Instead of using `tf.reduce_mean`, we could share some code with `ReduceLayer` (which can also calculate the mean, but correctly respects padded frames).

**Comment by albertz** [comment]
This should be configurable though, similar as in `BatchNormLayer`. And then the default should maybe be changed via new behavior version?
